src/ceph-daemon/ceph-daemon

```diff
@@ -1936,12 +1939,16 @@ if __name__ == "__main__":
             except Exception as e:
                 logger.debug('could not locate %s: %s' % (i, e))
         if not container_path:
-            raise RuntimeError('unable to locate any of %s' % CONTAINER_PREFERENCE)
+            raise CDError('unable to locate any of %s' % CONTAINER_PREFERENCE)
 
     if 'func' not in args:
         sys.stderr.write('No command specified; pass -h or --help for usage\n')
         sys.exit(1)
-    r = args.func()
+    try:
+        r = args.func()
+    except CDError as e:
```

Comment:
Catch `Exception` instead of `CDError`?

For example, when run as a non-privileged user:
```
$ ./ceph-daemon ls
Traceback (most recent call last):
  File "./ceph-daemon", line 1949, in <module>
    r = args.func()
  File "./ceph-daemon", line 1388, in command_ls
    legacy_dir=args.legacy_dir)
  File "./ceph-daemon", line 1402, in list_daemons
    for i in os.listdir(data_dir):
OSError: [Errno 13] Permission denied: '//var/lib/ceph'
```

Reply:
I don't think we should catch other exceptions because the Tracebacks are useful.. just seeing "Permission denied' makes this painful to understand.  For non-root users, for instance, we should just add an explicit check for whether we are root!

Reply:
I was thinking we could still make the traceback available using the `-v` arg ... 

https://github.com/ceph/ceph/pull/31886/files#diff-0d7e098eacbe52f3a7f86d925dfeff1fR1956-R1957

```
$ ./ceph-daemon ls
ERROR: [Errno 13] Permission denied: '//var/lib/ceph'


$ ./ceph-daemon -v ls
Traceback (most recent call last):
  File "./ceph-daemon", line 1957, in <module>
    r = args.func()
  File "./ceph-daemon", line 1389, in command_ls
    legacy_dir=args.legacy_dir)
  File "./ceph-daemon", line 1405, in list_daemons
    for i in os.listdir(data_dir):
__main__.Error: [Errno 13] Permission denied: '//var/lib/ceph'
```

Reply:
I like that you get a traceback with -v, but I also think we should include a traceback for any *unexpected* (non-Error) exception.  The current Error cases are all user input problems where a traceback isn't informative in any way, but anything else is not expected and a traceback is
 helpful.